### DISCUSSION TITLE
Support for (arm64) Nvidia JetPack >5.0

### DISCUSSION THREAD

**Discussion by namrata-jangid** [foxglove/community#667] [Q&A]
Hi, 

Our team (@kis-balazs) is having an issue with running Foxglove Studio on an Ubuntu 20.04 machine. We had opened an issue previously as well (https://github.com/foxglove/studio/issues/5564).

Attaching a few screenshots to further elaborate on the issue:

- Ubuntu version:
![ubuntu_version](https://user-images.githubusercontent.com/128360488/226284868-ccda728a-bb86-426b-937a-e7f3b8f78a79.png)

- Installing Foxglove Studio:
![install_foxglove](https://user-images.githubusercontent.com/128360488/226284920-51a22133-cafa-4be0-835d-7ad16d998d52.png)

- 'foxglove-studio' command does not run as expected:
![foxglove_command](https://user-images.githubusercontent.com/128360488/226285257-6b260588-8c02-4631-ba0e-84db59652bab.png)

Requesting your help on this. 

Thank you! :)

**Comment by foxhubber** [comment]
Internal tracking ticket: [FG-2531](https://linear.app/foxglove/issue/FG-2531/support-for-ubuntu-2004-[foxglovestudio-5591])

**Comment by jtbandes** [comment]
We're going to need more information to be able to help. The app works fine with my ubuntu 20.04 VM on arm64. Do you get any error messages?

<img width="1207" alt="image" src="https://user-images.githubusercontent.com/14237/226421740-9323c326-c4f4-49f2-8c26-78cfe174294d.png">

**Reply by james-rms** [reply]
@jtbandes when you ran this, was this from the snap or installed from the `.deb`?

**Reply by jtbandes** [reply]
The deb I believe.

**Comment by kis-balazs** [comment]
Hi!
Weird thing is exactly what you asked, there is nothing popping up as an error, and neither ```htop``` shows any significant move regarding a "foxglove-studio" executable (see last image)

We are running it on a Jetson Xavier, with JetPack 5.0.2, Ubuntu 20.04 as specified in the initial image. Can maybe this trigger such unusual behavior?

How could we figure this out better?

Thanks,
 Balazs

**Comment by jtbandes** [comment]
Are you able to run other apps built with Electron? https://github.com/electron/electron  You could also try https://www.electronjs.org/fiddle/. It sounds like there are some more logs you can enable as well: https://www.electronjs.org/docs/latest/api/command-line-switches#--enable-loggingfile

**Reply by kis-balazs** [reply]
Hi, I am back with logs. Please check comments :D

**Comment by kis-balazs** [comment]
Hi there!
We have run fiddle, and it has the same behavior, even though it shows up in the htop window, as being present... very weird behavior, and I believe it is an electron-related issue in this case. Do you have any idea how to solution this?
Please see screenshots below.

![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/25155919/227157746-34b3cc49-ca7f-4acc-aaa5-95c1b3eaa940.png)
![MicrosoftTeams-image](https://user-images.githubusercontent.com/25155919/227157752-ce80cecc-9c97-4ea4-b312-a41c2238f96a.png)
![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/25155919/227157757-ccc00214-a0f4-426f-a24e-59d5ff64945d.png)

**Comment by jtbandes** [comment]
Sorry, I've never experienced this problem before. I don't have any particular advice other than to search through the electron community to see if others have experienced this problem. I'm going to close this issue because we've determined the bug is not related to Foxglove Studio, but feel free to post further updates and we can reopen it if anything changes.

**Comment by kis-balazs** [comment]
Yeah well this wasn't useful. Then let's figure out the arm64 issues, in fact, arm64 architecture is not fully supported :) I think it's only correct not to induce people in error.

**Reply by amacneil** [reply]
I'm fairly sure it is not an arm64 issue, I run Foxglove Studio on ubuntu+arm64 pretty frequently without issue. I wonder if this is something specific to the Jetson hardware.

**Reply by kis-balazs** [reply]
This sounds like a very feasable issue, yet let's find a way to push for this in one way or another. Maybe we can propose a couple tests, or post some logs using a possible 'verbose' (is there one of such?) flag for Foxglove? If not, maybe it's time :)

**Reply by defunctzombie** [reply]
Sounds good. Let us know what you learn and we can figure out what adjustments to make for the arm64 build. Definitely starting with the base electron running is a good place cause if electron itself won't run then there's not much we'll be able to do in Studio.

**Comment by jtbandes** [comment]
Did you try running with `--enable-logging`? Any interesting log messages? https://www.electronjs.org/docs/latest/api/command-line-switches#--enable-loggingfile

**Reply by kis-balazs** [reply]
Let us try and will come back with more logs, running electronjs using the logging flag. Does Foxglove have such parameter(s) as well? :)

**Reply by kis-balazs** [reply]
See my answer below :)

**Comment by kis-balazs** [comment]
We tried running electron-fiddle in the following ways:
1.
![logging_flag](https://user-images.githubusercontent.com/25155919/230321175-6e8e06e0-84e6-4cac-a5da-883f70a2e097.png)
2.
![no-sandbox-logging window](https://user-images.githubusercontent.com/25155919/230321657-1363a6d7-e514-4e95-9bba-a7bc43e8e0d2.png)

 - resulting log file: [fiddle.txt](https://github.com/foxglove/community/files/11168065/fiddle.txt)

3.
![no-sandbox_and_gpu-logging](https://user-images.githubusercontent.com/25155919/230321388-2c81c21b-4196-4655-a99b-cd5e33e8b9d9.png)
 - resulting rendered window: ![no-sandbox_and_gpu-logging window](https://user-images.githubusercontent.com/25155919/230321458-c2562c46-e7b0-4689-93b3-7feaf17b62f0.png)
 - resulting log file: [fiddle_0.txt](https://github.com/foxglove/community/files/11168071/fiddle_0.txt)

---
Hope this helps shedding a little more light on the issue, we can try other parameters regarding this slightly unusal bus error, but I think this already is in the territory of needing a proper fix :)

**Reply by james-rms** [reply]
The EGL not initialized errors indicate that there might be some incompatibility or error with your graphics driver. Are you able to try again while monitoring `dmesg` for new error messages from the NVidia driver?

**Reply by james-rms** [reply]
People with similar issues seem to use `xhost +` as a first debugging step, see here: https://github.com/cypress-io/cypress/issues/26079 https://github.com/electron/electron/issues/36524
I'm really not familiar with the command, but it may be worth a shot to see if it changes anything.

**Reply by kis-balazs** [reply]
So, short update on the ```dmesg``` topic: **no** electron-fiddle related logs are showing up in the trace.

**Comment by kis-balazs** [comment]
So, we seem to be figured out the start-up:
- xhost + 
- sudo foxglove-studio --no-sandbox
---
Even though the start-up is successful, and we can work with Foxglove Studio, Ubuntu reports a problem.
I think we should investigate this, because the hacky solution is fun and all, but we would really need to have a sustainable solution for the future as well :)

**Reply by james-rms** [reply]
is the `sudo` neccessary? what happens when you run outside of sudo?

**Reply by kis-balazs** [reply]
Yes, it it mandatory.
Without ```sudo```, we're back to initial state, spin and no run :)
